src/db.rs

```diff
@@ -56,6 +57,47 @@ const KEEP_LOGS: usize = 16;
 /// Value is just a vector of bytes. Value sizes up to 4Gb are allowed.
 pub type Value = Vec<u8>;
 
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ValuePtr(Arc<Value>);
+
+impl ValuePtr {
+	pub fn value(&self) -> &Value {
+		&self.0
+	}
+}
+
+impl AsRef<[u8]> for ValuePtr {
+	fn as_ref(&self) -> &[u8] {
+		self.0.as_ref()
+	}
+}
+
+impl Borrow<[u8]> for ValuePtr {
+	fn borrow(&self) -> &[u8] {
+		self.value().borrow()
+	}
+}
+
+impl Borrow<Vec<u8>> for ValuePtr {
+	fn borrow(&self) -> &Vec<u8> {
+		self.value().borrow()
+	}
+}
+
+impl From<Value> for ValuePtr {
+	fn from(value: Value) -> Self {
+		Self(value.into())
+	}
+}
+
+impl<const N: usize> TryFrom<ValuePtr> for [u8; N] {
```

Comment:
is this function used?

Reply:
It's used in tests, let me mark it as `#[cfg(test)]`